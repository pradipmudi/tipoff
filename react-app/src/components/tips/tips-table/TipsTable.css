#tips-table thead {
    background-color: #e1f4f3;

}

#tips-table tbody{
    background-color: #ffffff;
}

.view-tip-icon {
    text-align: center !important;
}

.error{
    background-color: rgb(230, 188, 152) !important;
}

.warning{
    background-color: #f3f3b6  !important;
}

.success {
    background-color:rgb(204, 233, 178)  !important;
}

.unhandled {
    background-color:rgb(202, 136, 136)  !important;
}

.info {
    background-color:#a7c6d3  !important;
}

