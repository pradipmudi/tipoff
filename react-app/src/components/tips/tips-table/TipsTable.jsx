import React from 'react';
import { useHistory } from 'react-router-dom';
import Moment from 'react-moment';
import { Table, Content, Icon } from 'bloomer';

import UserAgentMini from '../../useragent/UserAgentMini';
import Tags from '../../common/tags/Tags';

import './TipsTable.css';

const TipsTable = ({tips}) => {
    
    const history = useHistory();

    const viewTip = (e, tipId) => { 
        e.preventDefault();
        history.push({
            pathname: '/tip/'+tipId
        });
    }
    return (
        <Content>
            <Table isBordered isStriped  id="tips-table">
                <thead >
                    <tr className = "header-row" >
                        <th className="is-hidden-touch">Event Type</th>
                        <th>UA</th>
                        <th>Event Title</th>
                        <th className="is-hidden-touch">Tags</th>
                        <th>Created On</th>
                        <th className="is-hidden-touch">View</th>
                    </tr>
                </thead>
                {tips && tips.length ? (
                    <tbody >
                    {tips.map((tip) => 
                        <tr key={tip.tip_id} 
                                className = {'tip-row '+tip.event_type} 
                                onClick={(e)=>{viewTip(e, tip.tip_id)}}
                                style={{cursor:"pointer"}}>
                            <td className="is-hidden-touch">{tip.event_type}</td>
                            <td><UserAgentMini ua={tip.user_agent} ></UserAgentMini></td>
                            <td>{tip.event_title}</td>
                            <td className="is-hidden-touch"><Tags tags={tip.tags} ></Tags> </td>
                            <td><Moment fromNow>{tip.created_on}</Moment></td>
                            <td className="is-hidden-touch view-tip-icon" ><Icon className='fas fa-file-alt' /></td>
                        </tr>
                    )}
                    </tbody>
                ) : (
                    <tbody>
                        <tr>
                            <td colSpan="6">This project hasn't received any Tips yet. You can refer to the Setup Guide to get started.</td>
                        </tr>
                    </tbody>
                )}
            </Table>
        </Content>
    )
};

export default TipsTable;
