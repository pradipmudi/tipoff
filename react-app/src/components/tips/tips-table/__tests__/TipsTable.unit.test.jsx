import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import TipsTable from '../TipsTable';

import TestUtils from '../../../../../../api-server/utils/test-utils';


describe("TipsTable", () => {

    const mockTips = TestUtils.getFakeTips(10,true);

    const props = {
        tips : mockTips
    }
    const wrapper = shallow(<TipsTable {...props} />);
    const rows = wrapper.find('.tip-row');

    it("matches snapshot", () => {
        const snap = renderer.create(<TipsTable />).toJSON();
        expect(snap).toMatchSnapshot();
    });

    it("renders without issues", () => {
        expect(wrapper.exists()).toEqual(true);
    });

    it("renders rows to match the number of tips ", () => {
        expect(rows.length).toEqual(mockTips.length);
    });

    it("renders td values to match the tip ", () => {
        const firstRowColumns = rows.first().find('td');
        let cols = firstRowColumns.map(col => {
            return col.html()
        });
        expect(firstRowColumns.length).toEqual(6);
        expect(cols[0]).toEqual(`<td>${mockTips[0].event_type}</td>`);
        expect(cols[1]).toContain('<time');
        expect(cols[2]).toEqual(`<td>${mockTips[0].event_title}</td>`);
        expect(cols[3]).toContain(`tag`);
        expect(cols[4]).toContain(`icon`);
        expect(cols[5]).toEqual(`<td>${mockTips[0].user.user_id}</td>`);
        
    });
});  
 