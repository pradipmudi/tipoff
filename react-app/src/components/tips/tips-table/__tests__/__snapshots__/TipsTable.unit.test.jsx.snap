// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipsTable matches snapshot 1`] = `
<div
  className="content"
>
  <table
    className="table is-bordered is-striped"
    id="tips-table"
  >
    <thead>
      <tr
        className="header-row"
      >
        <th
          className="is-hidden-touch"
        >
          Event Type
        </th>
        <th>
          UA
        </th>
        <th>
          Event Title
        </th>
        <th
          className="is-hidden-touch"
        >
          Tags
        </th>
        <th>
          Created On
        </th>
        <th
          className="is-hidden-touch"
        >
          View
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="6"
        >
          This project hasn't received any Tips yet. You can refer to the Setup Guide to get started.
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
